sstables/generation_type.hh

```diff
@@ -28,14 +28,22 @@ public:
     using int_t = int64_t;
 
 private:
-    int_t _value;
+    utils::UUID _value;
+
+    explicit constexpr generation_type(utils::UUID value) noexcept
+        : _value(value) {}
 
 public:
     generation_type() = delete;
 
-    explicit constexpr generation_type(int_t value) noexcept: _value(value) {}
-    constexpr int_t value() const noexcept { return _value; }
-
+    // use zero as the timestamp to differentiate from the regular timeuuid,
+    // and use the least_sig_bits to encode the value of generation identifier.
+    explicit constexpr generation_type(int_t value) noexcept
+        : _value(utils::UUID_gen::create_time(std::chrono::milliseconds::zero()), value) {}
+    constexpr int_t value() const noexcept {
+        assert(_value.timestamp() == 0);
```

Comment:
on_internal_error?

Reply:
we could use `sstlog` here, but since this is a header file providing a basic datatype for SSTable subsystem. i'd prefer using `assert()` in this case to avoid introducing the linkage to a symbol defined by `sstable.cc` which consumes the facilities provided by this file. once this file is complicated enough and by then, scylladb is not C++20 modularizied yet, we could consider extract the implementation into the .cc file and introduce yet another logger if we still think it's important to log this.

what do you think?

Reply:
I don't think pulling in the sstable logger is a problem here. You can just add an `extern logger sstlog;` and use it.

Reply:
Technically, it's okay. But it introduces yet another circular dependency.

Reply:
fair enough. will use `on_internal_error()` then.